### PR TITLE
Implement `pod show`

### PR DESCRIPTION
**Pull request by cruhland**

### CODE REVIEW

File: cli/dcoscli/marathon/main.py [hunk 1]
```diff
@@ -279,13 +279,17 @@ class MarathonSubcommand(object):
     :param create_marathon_client: a callable that returns an instance of
                                    marathon.Client
     :type create_marathon_client: collections.abc.Callable
+    :param event_emitter: sink for output events, e.g. messages to the console
+    :type event_emitter: emitting.Emitter
```
Comment (tamarrow) [inline cli/dcoscli/marathon/main.py]: I don't think we should have this here, we want all our emitting in the cli to stdout to be consistent.

**Comment by cruhland** [comment]
@tamarrow updated! I also modified the integration test for `pod show` so that it passes.

File: cli/dcoscli/marathon/main.py [hunk 2]
```diff
@@ -754,9 +754,9 @@ def deployment_watch(self, deployment_id, max_count, interval):
             else:
                 if 'TERM' in os.environ:
                     os.system('clear')
-            emitter.publish('Deployment update time: '
-                            '{} \n'.format(time.strftime("%Y-%m-%d %H:%M:%S",
-                                                         time.gmtime())))
+            formatted_time = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime())
+            message = 'Deployment update time: {} \n'.format(formatted_time)
```
Comment (tamarrow) [inline cli/dcoscli/marathon/main.py]: we can revert this change now.

File: cli/tests/integrations/test_marathon_pod.py [hunk 3]
```diff
@@ -121,9 +121,15 @@ def _assert_pod_remove(pod_id, extra_args):
     assert_command(cmd, returncode=0, stdout=b'', stderr=b'')
 
 
-def _assert_pod_show(pod_id, stdout):
+def _assert_pod_show(pod_id):
     cmd = _POD_SHOW_CMD + [pod_id]
-    assert_command(cmd, returncode=0, stdout=stdout, stderr=b'')
+    returncode, stdout, stderr = exec_command(cmd, env=None, stdin=None)
```
Comment (tamarrow) [inline cli/tests/integrations/test_marathon_pod.py]: this can just be `... = exec_command(cmd).

File: cli/tests/integrations/test_marathon_pod.py [hunk 4]
```diff
@@ -121,9 +121,15 @@ def _assert_pod_remove(pod_id, extra_args):
     assert_command(cmd, returncode=0, stdout=b'', stderr=b'')
 
 
-def _assert_pod_show(pod_id, stdout):
+def _assert_pod_show(pod_id):
     cmd = _POD_SHOW_CMD + [pod_id]
```
Comment (tamarrow) [inline cli/tests/integrations/test_marathon_pod.py]: lets move the `_POD_CMD` globals to the top of the file so theyre easy to find.

File: cli/tests/integrations/test_marathon_pod.py [hunk 5]
```diff
@@ -121,9 +121,15 @@ def _assert_pod_remove(pod_id, extra_args):
...
-    assert_command(cmd, returncode=0, stdout=stdout, stderr=b'')
+    returncode, stdout, stderr = exec_command(cmd, env=None, stdin=None)
+
+    assert returncode == 0
+    assert stderr == b''
+
+    pod_json = json.loads(stdout.decode('utf-8'))
+    assert pod_json['id'] == util.normalize_marathon_id_path(pod_id)
```
Comment (tamarrow) [inline cli/tests/integrations/test_marathon_pod.py]: how will this pass without the pods endpoint implemented?

File: cli/tests/unit/test_marathon_pod.py [hunk 6]
```diff
@@ -56,25 +68,58 @@ def resource_reader(path):
...
+
+
+def _assert_pod_show_propagates_exceptions_from_show_pod(exception):
+    marathon_client = create_autospec(marathon.Client)
+    marathon_client.show_pod.side_effect = exception
+
+    subcmd = main.MarathonSubcommand(
+        resource_reader=_unused_resource_reader,
+        create_marathon_client=lambda: marathon_client)
+    with pytest.raises(exception.__class__) as exception_info:
```
Comment (tamarrow) [inline cli/tests/unit/test_marathon_pod.py]: do we have a test already that a marathon failure results in the correct marathon error message?

File: tests/test_marathon.py [hunk 7]
```diff
@@ -80,6 +80,63 @@ def test_remove_pod_propagates_DCOSException():
...
+def test_show_pod_builds_rpc_correctly_1():
+    marathon_client, rpc_client = _create_fixtures()
+    marathon_client.show_pod('foo')
+    rpc_client.http_req.assert_called_with(http.get, 'v2/pods/foo')
+
+
+def test_show_pod_builds_rpc_correctly_2():
+    marathon_client, rpc_client = _create_fixtures()
+    marathon_client.show_pod('bar')
+    rpc_client.http_req.assert_called_with(http.get, 'v2/pods/bar')
```
Comment (tamarrow) [inline tests/test_marathon.py]: this is testing the same thing as the test above, no?

File: tests/test_marathon.py [hunk 8]
```diff
@@ -80,6 +80,63 @@ def test_remove_pod_propagates_DCOSException():
...
+    with pytest.raises(DCOSException) as exception_info:
+        marathon_client.show_pod('bad-req')
+
+    assert str(exception_info.value) == 'BOOM!'
+
+
+def test_show_pod_propagates_json_parsing_exception():
+    marathon_client, rpc_client = _create_fixtures()
+    mock_response = mock.create_autospec(requests.Response)
+    mock_response.json.side_effect = Exception('Bad parse')
```
Comment (tamarrow) [inline tests/test_marathon.py]: should return DcosException?

**Comment by tamarrow** [comment]
@cruhland a few small comments

File: cli/tests/integrations/test_marathon_pod.py (hunk 5 above)
Comment (cruhland) [inline cli/tests/integrations/test_marathon_pod.py]: It won't, but the tests are still marked as skipped. I enabled them for local testing to make this change. Just trying to prevent a big change once we actually enable these tests.

Comment (tamarrow) [inline cli/tests/integrations/test_marathon_pod.py]: 👍

File: cli/tests/unit/test_marathon_pod.py (hunk 6 above)
Comment (cruhland) [inline cli/tests/unit/test_marathon_pod.py]: Yes, but it depends on what your idea of a "correct marathon error message" is. [These tests](https://github.com/dcos/dcos-cli/blob/60ce2d47b899a4654faf801feb5b29fe3b0e1d3c/tests/test_marathon.py#L217-L359) cover the error messages for the statuses mentioned in the Marathon `/v2/pods` API docs: 400, 401, 403, 404, 409, and 422. Except for 400 and 409, the CLI doesn't do anything special with the error JSON, it just extracts some fields and prints them on the console. If we want nicer messages for specific statuses, let's figure out what they should be and we can write tests for them.

File: tests/test_marathon.py (hunk 7 above)
Comment (cruhland) [inline tests/test_marathon.py]: Yeah, it's the leftovers of TDD that I didn't refactor. I'll merge a few of the similar test cases together.

File: tests/test_marathon.py (hunk 8 above)
Comment (cruhland) [inline tests/test_marathon.py]: This test is actually showing that if `response.json()` results in an exception that _isn't_ `DCOSException`, then it will still propagate as expected.

File: cli/tests/unit/test_marathon_pod.py (hunk 6 above)
Comment (tamarrow) [inline cli/tests/unit/test_marathon_pod.py]: cool. just wanted to make sure we are catching errors from marathon appropriately- since they have their own code path

**Comment by cruhland** [comment]
@tamarrow a few small changes

**Comment by tamarrow** [comment]
@cruhland 1 tiny nit and then merge when tests pass.
